bin/pam_ball_trajectories.py

```diff
@@ -0,0 +1,209 @@
+        "info",
+        help="print information about the whole file, or a given group in the file.",
+    )
+    info.add_argument(
+        "--group", type=str, required=False, help="the group of trajectories"
+    )
+
+    # for adding the json files of the current folder
+    # to a hdf5 trajectory file
+    add_json = subparser.add_parser(
+        "add-json",
+        help="for saving in a new group all json trajectories present in the current directory",
+    )
+    add_json.add_argument(
+        "--group", type=str, required=True, help="the group of trajectories"
+    )
+    add_json.add_argument(
+        "--sampling_rate_us",
+        type=int,
+        required=True,
+        help="record sampling rate, in microseconds (int)",
+    )
+
+    # for adding the tennicam files of the current folder
+    # to a hdf5 trajectory file
+    add_tennicam = subparser.add_parser(
+        "add-tennicam",
+        help="for saving in a new group all tennicam present in the current directory",
+    )
+    add_tennicam.add_argument(
+        "--group", type=str, required=True, help="the group of trajectories"
+    )
+
+    # for removing a group from the hdf5 file
+    rm_group = subparser.add_parser(
+        "rm", help="for removing trajectories from the file"
+    )
+    rm_group.add_argument(
+        "--group", type=str, required=True, help="the group of trajectories"
+    )
+
+    # for translating all points in all the trajectories of the group
+    translate = subparser.add_parser(
+        "translate", help="translate all the positions of all trajectories of the group"
+    )
+    translate.add_argument(
+        "--group", type=str, required=True, help="the group of trajectories"
+    )
+    translate.add_argument(
+        "--coords", type=float, nargs=3, required=True, help="x y z coordinates (float)"
+    )
+
+    # parsing the arguments
+    args = parser.parse_args()
+
+    # all subparsers asks for an optional path. Using this path if provided,
+    # the default hdf5 path otherwise
+    if args.path:
+        hdf5_path = pathlib.Path(args.path)
+    else:
+        hdf5_path = bt.RecordedBallTrajectories.get_default_path(create=True)
+
+    # checking the file exists
+    if not pathlib.Path(hdf5_path).is_file():
+        raise FileNotFoundError("failed to find the file {}".format(hdf5_path))
+
+    # going ahead based on the arguments
+    if args.command == "info":
+        _info(hdf5_path, args.group)
+
+    elif args.command == "add-json":
+        _add_json(hdf5_path, args.group, args.sampling_rate_us)
+
+    elif args.command == "add-tennicam":
+        _add_tennicam(hdf5_path, args.group)
+
+    elif args.command == "rm":
+        _rm_group(hdf5_path, args.group)
+
+    elif args.command == "translate":
+        _translate(hdf5_path, args.group, args.coords)
+
+    else:
+        if args.command is None:
+            parser.print_help()
+            return
+        raise ValueError("unknown command: {}".format(args.command))
```

Comment:
You can make the "command" subparser above required, then this else case should not be necessary:

If only Python >=3.7 needs to be supported:
```
subparser = parser.add_subparsers(dest="command", required=True)
```
Alternative that also works for older Python versions:
```
subparser = parser.add_subparsers(dest="command")
subparser.required = True
```